frontend/src/pages/Registration.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react'
+import axios from "axios";
+import { Alert, Button, Form } from 'react-bootstrap';
+import apiUrls from "../constants/apiUrls"
+import axiosConfig from '../constants/axiosConfig';
+import User from '../interfaces/user';
+
+const client = axios.create(axiosConfig);
+
+function Registration() {
+  const emptyUser = {
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    is_admin: false,
+  };
+
+  const [user, setUser] = useState<User>(emptyUser)
+  const [error, setError] = useState<string>('')
+
+  async function registerUser() {
+    try {
+      await client.post(`${apiUrls.REGISTRATION}`, user)
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        setError(`${err.request.status} ${err.request.statusText}`);
+      }
+    }
+  }
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const {
+      target: { id, value, checked }
+    } = event
+
+    if (id === 'is_admin') {
+      setUser(prevState => ({
+        ...prevState,
+        [id]: checked
+      }))
+    } else {
+      setUser(prevState => ({
+        ...prevState,
+        [id]: value
+      }))
+    }
```

Comment:
I think you can optimize this if statement further. Focus on what field is being modified based sa condition then you can apply the conditions from there. 

Hint: you can use ternary operation for the condition in ID field